com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PollStageAppStatusExecution.java

```diff
@@ -114,26 +104,29 @@ protected ExecutionStatus executeStepInternal(DelegateExecution context) throws
         }
     }
 
-    private ExecutionStatus checkStagingState(DelegateExecution context, CloudFoundryOperations client, CloudApplication app,
+    private AsyncExecutionState checkStagingState(ExecutionWrapper execution, CloudFoundryOperations client, CloudApplication app,
         Pair<ApplicationStagingState, String> state) throws SLException {
 
         if (state._1.equals(ApplicationStagingState.FAILED)) {
             // Application staging failed
             String message = format(Messages.ERROR_STAGING_APP_2, app.getName(), state._2);
-            getStepLogger().error(message);
-            StepsUtil.saveAppLogs(context, client, recentLogsRetriever, app, LOGGER.getLoggerImpl(), processLoggerProviderFactory);
-            setRetryMessage(context, message);
-            return ExecutionStatus.LOGICAL_RETRY;
+            execution.getStepLogger().error(message);
+            StepsUtil.saveAppLogs(execution.getContext(), client, recentLogsRetriever, app, LOGGER,
+                execution.getProcessLoggerProviderFactory());
+            setType(execution, StepPhase.RETRY);
+            return AsyncExecutionState.ERROR;
         } else {
             // Application not staged yet, wait and try again unless it's a timeout
-            if (StepsUtil.hasTimedOut(context, () -> System.currentTimeMillis())) {
+            if (StepsUtil.hasTimedOut(execution.getContext(), () -> System.currentTimeMillis())) {
                 String message = format(Messages.APP_START_TIMED_OUT, app.getName());
-                getStepLogger().error(message);
-                StepsUtil.saveAppLogs(context, client, recentLogsRetriever, app, LOGGER.getLoggerImpl(), processLoggerProviderFactory);
-                setRetryMessage(context, message);
-                return ExecutionStatus.LOGICAL_RETRY;
+                execution.getStepLogger().error(message);
+                StepsUtil.saveAppLogs(execution.getContext(), client, recentLogsRetriever, app, LOGGER,
+                    execution.getProcessLoggerProviderFactory());
+                setType(execution, StepPhase.RETRY);
+                return AsyncExecutionState.ERROR;
             }
-            return ExecutionStatus.RUNNING;
+            setType(execution, StepPhase.POLL);
```

Comment:
Why not just StepsUtil.setStepPhase? This was a bit confusing to me at first and I had to go looking for this setType method in order to find out how it is different than the other.

Reply:
Again.. should be deleted.

Reply:
done